cmake/OpenCVDetectFindGflags.cmake

```diff
@@ -420,7 +420,7 @@ if (NOT GFLAGS_FOUND)
     set(CALLERS_CMAKE_FIND_LIBRARY_PREFIXES "${CMAKE_FIND_LIBRARY_PREFIXES}")
     # The empty string in this list is important, it represents the case when
     # the libraries have no prefix (shared libraries / DLLs).
-    set(CMAKE_FIND_LIBRARY_PREFIXES "lib" "" "${CMAKE_FIND_LIBRARY_PREFIXES}")
+    # LaurentBerger For Opencv PR 7700 set(CMAKE_FIND_LIBRARY_PREFIXES "lib" "" "${CMAKE_FIND_LIBRARY_PREFIXES}")
```

Comment:
Looks like the issue is caused by [this](https://ceres-solver.googlesource.com/ceres-solver/+/c5c6f559ab01de024b95b14432bf6a42378ae8ef) change. In case `CMAKE_FIND_LIBRARY_PREFIXES` is empty prior to processing `CALLERS_CMAKE_FIND_LIBRARY_PREFIXES` is also empty so check at line 88 became false and  `CMAKE_FIND_LIBRARY_PREFIXES` isn't reset.
I'm unsure whether `lib` prefix is really necessary for MSVS, but anyway it make sense either completely remove library prefix handling or make it to work right(IMO the better option). I.e. `CALLERS_CMAKE_FIND_LIBRARY_PREFIXES` could be unconditionally initialized prior to detection and therefore reset of `CMAKE_FIND_LIBRARY_PREFIXES` could be also done regardless of `CALLERS_CMAKE_FIND_LIBRARY_PREFIXES` state.